src/react_native/biometrics.cljs

```diff
@@ -0,0 +1,76 @@
+(ns react-native.biometrics
+  (:require
+    ["react-native-biometrics" :default biometrics]
+    [clojure.string :as string]
+    [oops.core :as oops]
+    [react-native.platform :as platform]
+    [schema.core :as schema]))
+
+(def ^:const biometric-error-not-available "biometric-not-available")
+(def ^:const biometric-error-not-enrolled "biometric-not-enrolled")
+
+(defn get-supported-type
+  "Returns a JS promise that resolves with the biometrics types supported by the
+  device, regardless of whether it's enabled or disabled.
+
+  Resolved values: `:Biometrics` `:FaceID` `:TouchID`"
+  []
+  (-> (biometrics.)
+      (.isSensorAvailable)
+      (.then (fn [result]
+               (let [type (-> result
+                              (oops/oget "biometryType")
+                              keyword)]
+                 type)))
+      (.catch (constantly nil))))
+
+(defn get-available
+  "Returns a JS promise that resolves to a boolean, which signifies whether
+  biometrics is enabled/disabled on the device."
+  []
+  (-> (biometrics.)
+      (.isSensorAvailable)
+      (.then (fn [result]
+               (oops/oget result "available")))))
+
+;; NOTE: the react-native-biometrics package error codes/messages differ across platforms.
+;; On android we get error messages while on iOS it's a stringified Obj-C error object.
+(defn- convert-auth-error-message
+  [code]
+  (letfn [(includes? [substring string]
+            (string/includes? string substring))]
+    (if platform/android?
+      (condp = code
+        "No fingerprints enrolled."      biometric-error-not-enrolled
```

Comment:
Kinda, we have specific user-facing popups for when biometrics is disabled and the error messages on android come in this way. Some with a period and some without. Since we're re-throwing them with proper codes and they're not used anywhere else, I'd prefer to keep them here WDYT?

Reply:
Oh for sure. I wasn't clear. I would extract the loose strings to private vars at the top of this namespace. As you said, we want them isolated. Magical strings have a tendency to be copied over and over, so having local vars should help discourage that (even in the same namespace).